naviqore_viewer/README.md

```diff
@@ -0,0 +1,22 @@
+# NAVIQORE Streamlit App
+
+## Setup
+
+### Install dependencies
+
+Run the command `poetry install` in your terminal to create a virtual environment with all requirements installed.
+
+### Configuration
+
+Create a `.env` file in the root directory. And add a line telling the application to which service host it should connect. In Local Development this is typically a Java Spring service running on `http://localhost:8080`.
+
+The required line in the `.env` file will then be: `NAVIQORE_HOST_URL=http://localhost:8080`
+
+_Note: you can also specify the `NAVIQORE_HOST_URL` in your operating system environment._
+
+## Starting the App
+
+To start the app run following command in your terminal from the root directory: `streamlit run naviqore_viewer/main.py`. _Note: The virtual environment should be active in your terminal session before starting the app._
```

Comment:
I had to use `poetry run streamlit run naviqore_viewer/main.py` since I only have streamlit installed in the poetry env.

Reply:
Addressed with a new Makefile approach (did some learning this morning :))

affef30

Reply:
Works 👍 

```bash
public-transit-viewer/naviqore_viewer on  feature/NAV-60-setup-basic-streamlit-frontend [!] is 📦 v0.1.0 via 🐍 v3.12.3 took 59s
❯ export NAVIQORE_HOST_URL=http://localhost:8080 && make run
Using virtualenv: /code/naviqore/public-transit-viewer/naviqore_viewer/.venv
poetry install --sync
Installing dependencies from lock file

No dependencies to install or update

Installing the current project: naviqore-viewer (0.1.0)
poetry run pre-commit install
pre-commit installed at .git/hooks/pre-commit
poetry run streamlit run naviqore_viewer/main.py

Collecting usage statistics. To deactivate, set browser.gatherUsageStats to false.


  You can now view your Streamlit app in your browser.

  Local URL: http://localhost:8501
  Network URL: http://192.168.67.3:8501
  External URL: http://185.12.131.147:8501
```